src/components/pages/observation/list/views/stats/table.tsx

```diff
@@ -46,15 +41,15 @@ export default function LifeListTable({ data, title, loadMoreUniqueSpecies, filt
               <tr key={`${specieName}`} className="fade">
                 <td>{specieName}</td>
 
-                <ExternalBlueLink
-                  href={
-                    filter?.groupName
-                      ? `/group/${filter.groupName}/observation/list?${address}&recoName=${specieName}`
-                      : `/observation/list?${address}&recoName=${specieName}`
-                  }
+                <LocalLink
+                  href={`/observation/list`}
+                  params={{ ...queryParams, view: "list", recoName: specieName }}
+                  prefixGroup={true}
                 >
-                  <td>{specieCount}</td>
-                </ExternalBlueLink>
+                  <ExternalBlueLink>
+                    <td>{specieCount}</td>
```

Comment:
`<LocalLink/>` and `<ExternalBlueLink/>` should be inside `<td/>` not outside please change that